openpype/hosts/houdini/plugins/publish/validate_unreal_staticmesh_naming.py

```diff
@@ -0,0 +1,108 @@
+                                   OptionalPyblishPluginMixin):
+    """Validate name of Unreal Static Mesh.
+
+    This validator checks if output node name has a collision prefix:
+            - UBX
+            - UCP
+            - USP
+            - UCX
+
+    This validator also checks if subset name is correct
+            - {static mesh prefix}_{Asset-Name}{Variant}.
+
+    """
+
+    families = ["staticMesh"]
+    hosts = ["houdini"]
+    label = "Unreal Static Mesh Name (FBX)"
+    order = ValidateContentsOrder + 0.1
+    actions = [SelectInvalidAction]
+
+    optional = True
+
+    @classmethod
+    def apply_settings(cls, project_settings, system_settings):
+        settings = (
+            project_settings["houdini"]["create"]["CreateUnrealStaticMesh"]
+        )
+        cls.collision_prefixes = settings["collision_prefixes"]
+        cls.static_mesh_prefix = settings["static_mesh_prefix"]
+
+    def process(self, instance):
+
+        if not self.is_active(instance.data):
+            return
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            nodes = [n.path() for n in invalid if isinstance(n, hou.Node)]
+            raise PublishValidationError(
+                "See log for details. "
+                "Invalid nodes: {0}".format(nodes)
+            )
+
+    @classmethod
+    def get_invalid(cls, instance):
+
+        invalid = []
+
+        rop_node = hou.node(instance.data["instance_node"])
+        output_node = instance.data.get("output_node")
+        if output_node is None:
+            cls.log.debug(
+                "No Output Node, skipping check.."
+            )
+            return
+
+        if not rop_node.evalParm('buildfrompath'):
+            # This validator doesn't support naming check if
+            # building hierarchy from path' is used
+            cls.log.info(
+                "Using 'Build Hierarchy from Path Attribute', skipping check.."
+            )
+            return
+
+        # Check nodes names
+        all_outputs = get_output_children(output_node, include_sops=False)
+        for output in all_outputs:
+            for prefix in cls.collision_prefixes:
+                if output.name().startswith(prefix):
+                    invalid.append(output)
+                    cls.log.error(
+                        "Invalid node name: Node '%s' "
+                        "includes a collision prefix '%s'",
+                        output.path(), prefix
+                    )
+                    break
+
+        # Check subset name
+        subset_name = "{}_{}{}".format(
+            cls.static_mesh_prefix,
+            instance.data["asset"],
+            instance.data.get("variant", "")
+        )
+
+        if instance.data.get("subset") != subset_name:
+            invalid.append(rop_node)
+            cls.log.error(
+                "Invalid subset name on rop node '%s' should be '%s'.",
+                rop_node.path(), subset_name
+            )
```

Comment:
So what is currently ensuring that on create the subset name is like that? :) I feel like the Creator class is missing an override to its `get_subset_name` method if this is really the rule for these instances.

Reply:
I think it's automated via dynamic data

Reply:
No, you shared a screenshot just now that the family is overridden by name due to the templating system in settings - so the default `get_subset_name` is already relying on that as it should. 👍 Anyway, I think the validator should actually just ensure it matches the template settings - not this hardcoded default?